adhocracy4/ratings/static/ratings/rating_api.js

```diff
@@ -0,0 +1,99 @@
+import api from '../../../static/api'
+
+/**
+ * @param {number} number - the rating value, can be -1, 0 or 1
+ * @param {string} objectId - the object to be rated
+ * @param {string} contentTypeId - the content type id of the object
+
+ * @returns {Promise<Array>} - an array with the rating data (number of negative
+ *    and positive ratings) and the user rating data (value and id and userHasRating)
+ *    and the complete response data from the API.
+ */
+export async function createRating (number, objectId, contentTypeId) {
+  try {
+    const data = await api.rating.add({
+      urlReplaces: {
+        objectPk: objectId,
+        contentTypeId
+      },
+      value: number
+    })
+
+    return [{
+      positive: data.meta_info.positive_ratings_on_same_object,
+      negative: data.meta_info.negative_ratings_on_same_object
+    }, {
+      userRating: data.meta_info.user_rating_on_same_object_value,
+      userHasRating: true,
+      userRatingId: data.id
+    }, data]
+  } catch (error) {
+    if (error.status === 400 &&
+        error.responseJSON.length === 1 &&
+        Number.isInteger(parseInt(error.responseJSON[0]))
+    ) {
+      const userRatingId = parseInt(error.responseJSON[0])
```

Comment:
What's the error case here? Trying to add a rating instead of modifying it? E.g. if the user rates in a different tab and then tries to vote in another one (which is not aware of the first vote yet)?